include/CL/sycl/buffer/detail/accessor.hpp

```diff
@@ -429,10 +429,10 @@ class accessor : public detail::debug<accessor<T,
 
     /* Create the OpenCL buffer and copy in data from the host if in
        read mode */
-    cl_buf = { task->get_queue()->get_boost_compute().get_context(),
-               get_size(),
-               flags,
-               is_read_access() ? array.data() : 0 };
+    cl_buf = boost::compute::buffer{ task->get_queue()->get_boost_compute().get_context(),
```

Comment:
Yes, well for MSVC that is. Problem is that `cl_buf` is a `boost::optional<boost::compute::buffer>`. The uniform initialization however does not initialize optional itself, but buffer, which is only one of optionals CTORs. MSVC does not recognize this (truth be told, I don't even know if it's valid). As far as I saw from `optional`s [header](http://www.boost.org/doc/libs/1_58_0/libs/optional/doc/html/optional/reference/header__boost_optional_optional_hpp_.html#boost_optional.reference.header__boost_optional_optional_hpp_.header_optional_optional), it does not inherit the CTORs of the stored type.

Writing out `boost::optional<boost::compute::buffer>` will invoke the `T&&` CTOR of `optional`, which is essentially the next best thing after emplace.